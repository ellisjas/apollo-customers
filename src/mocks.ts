import { ZellerCustomers } from "./API";

export const customer: ZellerCustomers = {
  id: '3457356745',
  email: 'jasmine@example.com',
  name: 'Jasmine',
  role: 'ADMIN',
  __typename: 'ZellerCustomers',
}

