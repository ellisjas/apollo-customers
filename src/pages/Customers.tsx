import React, { useState } from 'react'
import { useQuery } from '@apollo/client'
import { CustomerList, RadioButtons } from '../components'
import { ListZellerCustomers } from '../queries'
import CircularProgress from '@mui/material/CircularProgress'

const Customers: React.FC = () => {
  const [customerRole, setCustomerRole] = useState('ADMIN')

  const { data, loading, error } = useQuery(ListZellerCustomers, {
    variables: {
      filter: {
        role: {
          eq: customerRole,
        },
      },
    },
  })

  const handleSelectCustomerRole = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setCustomerRole(event.target.value)
  }

  const renderContent = () => {
    if (error) {
      console.error(error)
      return <h1>Error!</h1>
    }

    if (loading)
      return (
        <div>
          <CircularProgress />
        </div>
      )

    return (
      <CustomerList
        customers={data.listZellerCustomers.items}
        role={customerRole}
      />
    )
  }

  return (
    <>
      <h1>Customers</h1>
      {renderContent()}
    </>
  )
}

export default Customers
