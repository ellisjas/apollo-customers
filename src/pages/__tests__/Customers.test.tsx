import { MockedProvider } from '@apollo/client/testing'
import { render } from '@testing-library/react'
import React from 'react'
import { getQueryMocks } from '../../mocks'
import Customers from '../Customers'

const role = 'ADMIN'

describe('pages: <Customers />', () => {
  it('renders', async () => {
    const { container } = render(
      <MockedProvider mocks={getQueryMocks(role)} addTypename={false}>
        <Customers />
      </MockedProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
