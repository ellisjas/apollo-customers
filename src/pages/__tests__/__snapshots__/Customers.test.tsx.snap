// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages: <Customers /> renders 1`] = `
<div>
  <div
    class="sc-pyfCe iIWjnJ"
  >
    <h1
      class="sc-bcXHqe pbKPE"
    >
      User Types
    </h1>
    <div
      class="sc-gKPRtg kQxcki"
    >
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <div
          aria-labelledby="demo-radio-buttons-group-label"
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd sc-iBYQkv ljGVhS css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="radio-buttons-group"
                type="radio"
                value="ADMIN"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Admin
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd sc-iBYQkv ljGVhS css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="radio-buttons-group"
                type="radio"
                value="MANAGER"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Manager
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      data-testid="loading-state"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
